test/e2e/lib/eventsRegistry.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  AnyServerEvent,
+  isRumErrorEvent,
+  isRumResourceEvent,
+  isRumUserActionEvent,
+  isRumViewEvent,
+  ServerInternalMonitoringMessage,
+  ServerLogsMessage,
+  ServerRumEvent,
+} from './serverTypes'
+
+type IntakeType = 'logs' | 'rum' | 'internalMonitoring'
+
+interface Entry {
+  type: IntakeType
+  event: AnyServerEvent
+}
+
+export class EventRegistry {
+  private db: Entry[] = []
+
+  push(type: IntakeType, event: AnyServerEvent) {
+    this.db.push({ type, event })
+  }
+
+  get all() {
+    return this.db.map((entry) => entry.event)
+  }
+
+  get logs() {
+    return this.getEvents('logs')
+  }
+
+  get rum() {
+    return this.getEvents('rum')
+  }
```

Comment:
I used a single list to keep the events order regardless of their type. By using multiple lists, we'll lose this information, but it should be fine since we don't have tests on this for now!